### PR TITLE
[dashboard] Use new Google Sign In button.

### PR DESCRIPTION
**Pull request by ditman**
This PR:

1. Renames the old `SignInButton` widget to `UserSignIn`. This was done because the old SignInButton did much more than rendering a button (it also rendered the "signed-in" status of the user).
2. Adds a new `SignInButton` widget, which has two variants:
    * For native, it is the old implementation
    * For web, it uses the new renderButton widget (this required adding two new "any" dependencies to pubspec)
3. Tweaks the authentication service a little bit (removed unused scopes, simplified some checks)
4. Fixes an issue where the top navbar wouldn't update after sign-in/sign-out. (Not sure if there's a github issue for this!)
5. Regenerates mockito mocks, and helps others do the same down the line :)
6. Tweaks to the index_page and sidebar.

## Issues

* Fixes: https://github.com/flutter/flutter/issues/121275
* Fixes: https://github.com/flutter/flutter/issues/103071

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dashboard/lib/widgets/sign_in_button/sign_in_button_web.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import 'package:google_sign_in_platform_interface/google_sign_in_platform_interface.dart';
+import 'package:google_sign_in_web/google_sign_in_web.dart';
+
+/// Widget that users can click to initiate the Sign In process.
+class SignInButton extends StatelessWidget {
+  const SignInButton({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return (GoogleSignInPlatform.instance as GoogleSignInPlugin).renderButton();
```
Comment (ditman) [inline dashboard/lib/widgets/sign_in_button/sign_in_button_web.dart:16]: We can tweak the looks of the button, see a configurable demo here:

https://www.dit-gis-button-test.web.app

File: dashboard/test/utils/mocks.mocks.dart [hunk 2]
```diff
@@ -1,299 +1,800 @@
-// Mocks generated by Mockito 5.0.15 from annotations
+// Mocks generated by Mockito 5.4.0 from annotations
```
Comment (ditman) [inline dashboard/test/utils/mocks.mocks.dart:1]: It's been a while!

File: dashboard/web/index.html [hunk 3]
```diff
@@ -6,7 +6,7 @@
 <head>
   <meta charset="UTF-8">
   <meta name="google-signin-client_id" content="308150028417-vlj9mqlm3gk1d03fb0efif1fu5nagdtt.apps.googleusercontent.com">
-  <title>Build Dashboard</title>
+  <title>Flutter Build Dashboard — Cocoon</title>
```
Comment (ditman) [inline dashboard/web/index.html:9]: To improve SEO (for myself) :P

**Review by ditman — COMMENTED** [review]
Hello new sign-in button!

File: dashboard/test/utils/mocks.mocks.dart
Comment (ditman) [inline dashboard/test/utils/mocks.mocks.dart:1]: I had to re-format this file after re-generating this file (it seems mockito generates at 80 cols, but you validate at 120). Please consider skipping *.mocks.dart from the formatting checks!

**Comment by ricardoamador** [comment]
Nice! Thanks for doing this David! If I can clone your branch I can do a test deploy before merging this.

File: dashboard/regen_mocks.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# Copyright 2019 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+dart pub get
```
Comment (CaseyHillers) [inline dashboard/regen_mocks.sh:6]:
```suggestion
flutter pub get
```

**Review by CaseyHillers — APPROVED** [review]

File: dashboard/regen_mocks.sh (hunk 4 above)
Comment (ditman) [reply dashboard/regen_mocks.sh]: I was getting a deprecation warning, but it probably came from the "flutter pub run" line below (now dart run).

**Comment by ditman** [comment]
> Nice! Thanks for doing this David! If I can clone your branch I can do a test deploy before merging this.

@ricardoamador you should be able to fetch from my remote and clone the branch, I think? I marked the checkbox of "collaborators can do whatever they want" while creating the PR (you should be even able to push to the branch if you want to!)

**Comment by ricardoamador** [comment]
Cool I will check it out.

File: dashboard/test/utils/mocks.mocks.dart
Comment (ditman) [reply dashboard/test/utils/mocks.mocks.dart:1]: Alternatively, this could run `dart format` with the right parameters after mocks are regenerated :/

**Comment by ricardoamador** [comment]
Okay first run went like this:
```
Restarted application in 191ms.
[GSI_LOGGER-TOKEN_CLIENT]: Instantiated.
[GSI_LOGGER]: The given origin is not allowed for the given client ID.
[GSI_LOGGER]: Prompt will not be displayed
[google_sign_in_web] Error on CredentialResponse:
[google_sign_in_web] Removing error from `userDataEvents`:
[GSI_LOGGER]: Processing click for button: gsi_283171_880200.

```
Looks like I will need to get my feet wet.

**Comment by ricardoamador** [comment]
Okay validated the change and it now works with the google login.
image

**Review by ricardoamador — APPROVED** [review]
LGTM!

**Comment by ditman** [comment]
Thanks @CaseyHillers @ricardoamador for the super quick review!

**Pull request merged**